vowpalwabbit/core/src/parse_args.cc

```diff
@@ -408,6 +408,51 @@ input_options parse_source(VW::workspace& all, options_i& options)
                      "hashed as A^B^C."))
       .add(make_option("flatbuffer", parsed_options.flatbuffer)
                .help("Data file will be interpreted as a flatbuffer file")
+               .experimental())
+      .add(make_option("csv", parsed_options.csv).help("Data file will be interpreted as a CSV file").experimental())
+      .add(make_option("csv_separator", all.csv_separator)
```

Comment:
is there any way to make these variables live in the parser rather than in all (maybe through the `input_options` object)? Either way, can we put them into a single struct to keep them organized?

Reply:
Just move all the csv options to a struct under VW/parsers/csv

Reply:
There is an interface (https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/vowpalwabbit/core/include/vw/core/input_parser.h) which exists on in the `all` object for exactly this purpose.

Reply:
@jackgerrits Ok, cool, will check that for details.

Reply:
Hi @jackgerrits, I would like to ask what is that interface for exactly? Does that mean we have to re-realize the options parser? As I can't find a realization of that interface, could you explain that in details? If it's for the options parser, I guess keep it just as the current way would be better as I think we shall make the CSV parser the internal one instead of the external parser?

Reply:
The point of that interface is to allow type erasure of any input parser that needs to be stored inthe `all` object. That way, all parser specific state can live in the instance as it is not relevant to the rest of the codebase.

This is an example of it being used: https://github.com/VowpalWabbit/reinforcement_learning/blob/3e832a8cc990ea0d9ca54a843af83730fdfd6524/external_parser/parse_example_external.cc#L183

However, in this case it would be done in the area where the csv parser is "activated"

Reply:
@jackgerrits Thanks for your explanation. Has used that interface in my last commit.